### PR TITLE
(0.8.0) Drop support for prescribing T and S in kelp and gas exchange models

### PR DESCRIPTION
**Pull request by jagoosw**
I've been meaning to drop support for prescribing temperature and salinity in the gas exchange and kelp models as it made a mess of the code and is a hangover from some of our column experiments which isn't needed since they can just be put in the model as a function field much more cleanly.

I have also removed the mess that was how gas exchange model functions were setup since I found out you can subtype Function and it solved the prior issues.

I have also changed the eady example to have a T and S dependant buoyancy model as it is now essential for kelp models (and makes sense anyway).

This breaks the API for `GasExchange` and `SLatissima`.

Depends on [Oceananigans PR #3323](https://github.com/CliMA/Oceananigans.jl/pull/3323)

### CODE REVIEW

**Comment by jagoosw** [comment]
TODO: Remove erroneous changes to `Project.toml`

File: Project.toml [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-version = "0.7.0"
+version = "0.8.0"
 
 [deps]
 Adapt = "79e6a3ab-5dfb-504d-930d-738a2a938a0e"
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
+CairoMakie = "13f3f980-e62b-5c42-98c6-ff1f3baf88f0"
+DataDeps = "124859b0-ceae-595e-8997-d05f6a7a8dfe"
+Documenter = "e30172f5-a6a5-5a46-863b-614d45cd2de4"
+Interpolations = "a98d9a8b-a2ab-59e6-89dd-64a1c18fca59"
```
Comment (navidcy) [inline Project.toml:12]: Are these pkgs added as dependency on purpose?

Comment (jagoosw) [reply Project.toml:12]: No, will remove

**Comment by navidcy** [comment]
Since https://github.com/CliMA/Oceananigans.jl/pull/3323 was merged, I added a compat for Oceananigans v0.89.2.

**Comment by jagoosw** [comment]
everything seems to work again now

File: Project.toml [hunk 2]
```diff
@@ -13,13 +14,12 @@ StructArrays = "09ab397b-f2b6-538f-b94a-2f83cf4a842a"
 
 [compat]
 Adapt = "3"
+Documenter = "0.27"
```
Comment (navidcy) [inline Project.toml:17]: seems like this is not up to date with latest `main` that includes a compat for Documenter v1? could you merge main to this branch?

File: src/Models/Individuals/SLatissima.jl [hunk 3]
```diff
@@ -21,10 +21,10 @@ module SLatissimaModel
 using Roots, KernelAbstractions
 using OceanBioME.Particles: BiogeochemicalParticles, get_node
 using Oceananigans.Units
-using Oceananigans: CPU, Center
+using Oceananigans: CPU, Center, NonhydrostaticModel, HydrostaticFreeSurfaceModel
 using Oceananigans.Architectures: arch_array, device, architecture
-using Oceananigans: NonhydrostaticModel, HydrostaticFreeSurfaceModel
 using Oceananigans.Biogeochemistry: required_biogeochemical_tracers, biogeochemical_auxiliary_fields
+using Oceananigans.Utils: SumOfArrays
```
Comment (navidcy) [inline src/Models/Individuals/SLatissima.jl:27]: why is this needed?

**Review by navidcy — APPROVED** [review]
lgtm

just a few suggestions!

File: src/Models/Individuals/SLatissima.jl [hunk 4]
```diff
@@ -118,15 +116,15 @@ Keyword Arguments
 
 - `architecture`: the architecture to adapt arrays to
 - `growth_rate_adjustement`, ..., `exudation_redfield_ratio`: parameter values
-- `pescribed_velocity`, `pescribed_temperature` and `pescribed_salinity`: functions for the relative velocity, temperature and salinity in the form `f(x, y, z, t)`
+- `prescribed_velocity`: functions for the relative velocity, temperature and salinity in the form `f(x, y, z, t)`
```
[the remark this answers is not in the document]
Comment (jagoosw) [reply src/Models/Individuals/SLatissima.jl:119]: Oh yeah I didn't update the docstring correctly
```suggestion
- `prescribed_velocity`: functions for the relative velocity `f(x, y, z, t)`
```

I do think this is something we would want to get rid of at some point but I think it might still be essential in some cases

File: src/Models/Individuals/SLatissima.jl (hunk 3 above)
Comment (jagoosw) [reply src/Models/Individuals/SLatissima.jl:27]: Ah yes it is not anymore 
```suggestion
```

File: src/Models/Individuals/SLatissima.jl [hunk 5]
```diff
@@ -21,10 +21,10 @@ module SLatissimaModel
 using Roots, KernelAbstractions
 using OceanBioME.Particles: BiogeochemicalParticles, get_node
 using Oceananigans.Units
-using Oceananigans: CPU, Center
+using Oceananigans: CPU, Center, NonhydrostaticModel, HydrostaticFreeSurfaceModel
```
[the remark this answers is not in the document]
Comment (jagoosw) [reply src/Models/Individuals/SLatissima.jl:24]: I think this is a left over from something else
```suggestion
```

File: src/Models/Individuals/SLatissima.jl [hunk 6]
```diff
@@ -348,8 +342,11 @@ function update_lagrangian_particle_properties!(particles::SLatissima, model, bg
 
     update_particle_properties_kernel! = _update_lagrangian_particle_properties!(device(arch), workgroup, worksize)
 
+    tracer_fields = merge(model.tracers, model.auxiliary_fields)
+                    #NameTuple(field_name => total_field(field_name, field, model.background_fields.tracers) for (field_name, field) in pairs(merge(model.tracers, model.auxiliary_fields)))
```
[the remark this answers is not in the document]
Comment (jagoosw) [reply src/Models/Individuals/SLatissima.jl:346]: No don't need it anymore
```suggestion
```

**Review by jagoosw — COMMENTED** [review]
Didn't mean to make this comment but can't delete it now

**Comment by jagoosw** [comment]
> lgtm
> 
> just a few suggestions!

Thank you!
